overlord/patch/patch.go

```diff
@@ -40,45 +49,147 @@ func Init(s *state.State) {
 		panic("internal error: expected empty state, attempting to override patch-level without actual patching")
 	}
 	s.Set("patch-level", Level)
+
+	if s.Get("patch-sublevel", new(int)) != state.ErrNoState {
+		panic("internal error: expected empty state, attempting to override patch-sublevel without actual patching")
+	}
+	s.Set("patch-sublevel", Sublevel)
+}
+
+// applySublevelPatches applies all sublevel patches for given level, starting
+// from firstSublevel index.
+func applySublevelPatches(level, firstSublevel int, s *state.State) error {
+	for sublevel := firstSublevel; sublevel < len(patches[level]); sublevel++ {
+		if sublevel > 0 {
+			logger.Noticef("Patching system state level %d to sublevel %d...", level, sublevel)
+		}
+		err := applyOne(patches[level][sublevel], s, level, sublevel)
+		if err != nil {
+			logger.Noticef("Cannot patch: %v", err)
+			return fmt.Errorf("cannot patch system state to level %d, sublevel %d: %v", level, sublevel, err)
+		}
+	}
+	return nil
+}
+
+// maybeResetSublevelForLevel60 checks if the previously installed core revision was a patch 6 sublevel 0
+// i.e. unaware of sublevels and the need for resetting sublevel state and if so, resets sublevel to 0
+// to re-apply sublevel patches.
+func maybeResetSublevelForLevel60(s *state.State, sublevel *int) error {
+	s.Lock()
+	defer s.Unlock()
+
+	core, err := snapstate.CoreInfo(s)
```

Comment:
That's bogus. We'll need to dig into the state manually instead. A large reason to have a patch system like this is being able to cook code that can assume the data format in the backend is current. That's why the very first thing we do when we start snapd is patch the database, and that's why we won't need hacks in snapstate and everywhere else to locally consider whether the backend was patched or not.

This is true here, and this is true for every patch in this package.

Reply:
Fixed in the prerequisite PR.